slsDetectorServers/mythen3DetectorServer/slsDetectorFunctionList.c

```diff
@@ -333,6 +388,226 @@ int configureMAC(uint32_t destip, uint64_t destmac, uint64_t sourcemac, uint32_t
 	return OK;
 }
 
+/* pattern */
+
+uint64_t readPatternWord(int addr) {
+    // error (handled in tcp)
+    if (addr < 0 || addr >= MAX_PATTERN_LENGTH) {
+        FILE_LOG(logERROR, ("Cannot get Pattern - Word. Invalid addr 0x%x. "
+                "Should be between 0 and 0x%x\n", addr, MAX_PATTERN_LENGTH));
+        return -1;
+    }
+
+    FILE_LOG(logINFORED, ("  Reading (Executing) Pattern Word (addr:0x%x)\n", addr));
+    uint32_t reg_lsb = PATTERN_STEP0_LSB_REG + addr; // the first word in RAM as base plus the offset of the word to write (addr)
+	uint32_t reg_msb = PATTERN_STEP0_MSB_REG + addr;
+
+    // read value
+    uint64_t retval = get64BitReg(reg_lsb, reg_msb);
+    FILE_LOG(logDEBUG1, ("  Word(addr:0x%x) retval: 0x%llx\n", addr, (long long int) retval));
+
+    return retval;
+}
+
+uint64_t writePatternWord(int addr, uint64_t word) {
+    // get
+    if (word == -1)
+        return readPatternWord(addr);
+
+    // error (handled in tcp)
+    if (addr < 0 || addr >= MAX_PATTERN_LENGTH) {
+        FILE_LOG(logERROR, ("Cannot set Pattern - Word. Invalid addr 0x%x. "
+                "Should be between 0 and 0x%x\n", addr, MAX_PATTERN_LENGTH));
+        return -1;
+    }
+
+    FILE_LOG(logINFO, ("Setting Pattern Word (addr:0x%x, word:0x%llx)\n", addr, (long long int) word));
+    uint32_t reg_lsb = PATTERN_STEP0_LSB_REG + addr; // the first word in RAM as base plus the offset of the word to write (addr)
+	uint32_t reg_msb = PATTERN_STEP0_MSB_REG + addr;
+
+    // write word
+    set64BitReg(word, reg_lsb, reg_msb);
+    FILE_LOG(logDEBUG1, ("  Wrote word. PatternIn Reg: 0x%llx\n", get64BitReg(reg_lsb, reg_msb)));
+
+    return word;
```

Comment:
Instead of return word, I would do "return readPatternword(addr)", to ensure it returns what it is really in there. In the ctb List.c, I was not calling readPatternWord as it would execute the pattern, but I guess that is not the case here right?